pkg/nmstate/nmstate.go

```diff
@@ -1 +1,223 @@
 package nmstate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+
+	nmstateV1 "github.com/nmstate/kubernetes-nmstate/api/v1"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Builder provides struct for the NMState object containing connection to
+// the cluster and the NMState definitions.
+type Builder struct {
+	Definition *nmstateV1.NMState
+	Object     *nmstateV1.NMState
+	apiClient  *clients.Settings
+	errorMsg   string
+}
+
+// NewBuilder creates a new instance of Builder.
+func NewBuilder(apiClient *clients.Settings, name string) *Builder {
+	glog.V(100).Infof("Initializing new NMState structure with the name: %s", name)
+
+	builder := Builder{
+		apiClient: apiClient,
+		Definition: &nmstateV1.NMState{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the NMState is empty")
+
+		builder.errorMsg = "NMState 'name' cannot be empty"
+	}
+
+	return &builder
+}
+
+// Exists checks whether the given NMState exists.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if NMState %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	if err != nil {
+		glog.V(100).Infof("Failed to collect NMState object due to %s", err.Error())
+	}
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Get returns NMState object if found.
+func (builder *Builder) Get() (*nmstateV1.NMState, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Collecting NMState object %s", builder.Definition.Name)
+
+	nmstate := &nmstateV1.NMState{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{Name: builder.Definition.Name}, nmstate)
+
+	if err != nil {
+		glog.V(100).Infof("NMState object %s doesn't exist", builder.Definition.Name)
+
+		return nil, err
+	}
+
+	return nmstate, err
+}
+
+// Create makes a NMState in the cluster and stores the created object in struct.
+func (builder *Builder) Create() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the NMState %s", builder.Definition.Name)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
+// Delete removes NMState object from a cluster.
+func (builder *Builder) Delete() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting the NMState object %s",
+		builder.Definition.Name,
+	)
+
+	if !builder.Exists() {
```

Comment:
Please remove this check. We basically don't care if NS exist in case if we want to remove it. If it's important you can leave the log message but no need to return error.

Reply:
Just add something like:

	if !builder.Exists() {
		return nil
	}

Reply:
removed.